packages/components/src/mobile/image-viewer/index.native.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { View, PanResponder, Dimensions, ImageBackground, Image } from 'react-native';
+import Modal from 'react-native-modal';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+import { calculateFullscreenImageSize } from './utils';
+
+// Default Image ratio 4:3
+const IMAGE_ASPECT_RATIO = 4 / 3;
+
+class ImageViewer extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			width: undefined,
+			height: undefined,
+		};
+
+		this.onDimensionsChange = this.onDimensionsChange.bind( this );
+	}
+	
+	componentDidMount() {
+		Dimensions.addEventListener("change", this.onDimensionsChange);
+		this.fetchImageSize();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.url !== prevProps.url ) {
+			this.fetchImageSize();
+		}
+	}
+
+    componentWillUnmount() {
```

Comment:
Minor, but we should be using tabs only. We have a pre-commit hook for linting and other checks. You'll probably need the dependencies installed in `gutenberg` for that to work (`npm install`)

Reply:
Yea I noticed I have a couple styling hiccups in here. I'll be sure to fix those.